lib/Config_option.ml

```diff
@@ -188,46 +188,91 @@ module Make (C : CONFIG) = struct
     store := Pack opt :: !store ;
     opt
 
-  type removed_value = {name: string; version: string; msg: string}
-
-  let removed_value ~name ~version ~msg = {name; version; msg}
-
-  let removed_values ~names ~version ~msg =
-    List.map names ~f:(fun name -> removed_value ~name ~version ~msg)
-
-  let add_removed_values values conv =
-    let parse s =
-      match List.find values ~f:(fun {name; _} -> String.equal name s) with
-      | Some {name; version; msg} ->
-          Format.kasprintf
-            (fun s -> Error (`Msg s))
-            "value `%s` has been removed in version %s. %s" name version msg
-      | None -> Arg.conv_parser conv s
-    in
-    Arg.conv (parse, Arg.conv_printer conv)
+  module Value = struct
+    module Ok = struct
+      type 'a t = string * 'a * string * [`Valid | `Deprecated of deprecated]
+
+      let valid ~name ~doc value = (name, value, doc, `Valid)
```

Comment:
What about `make` rather than `valid` ? `deprecated` can be turned into `make_deprecated` too.

Reply:
done